lib/subnet-routes.js

```diff
@@ -0,0 +1,79 @@
+/** @import { RequestWithSubnet } from './typings.js' */
+/** @import { FastifyInstance, FastifyReply } from 'fastify' */
+import { subnetPreHandlerHook } from './request-helpers.js'
+
+/**
+ * Define the subnet routes
+ * @param {FastifyInstance} app
+ */
+export const subnetRoutes = (app) => {
+  app.addHook('preHandler', subnetPreHandlerHook)
+
+  app.post(
+    '/:subnet/measurement',
+    {
+      schema: {
+        body: {
+          type: 'boolean'
+        }
+      }
+    },
+    /**
+     * @param {RequestWithSubnet} request
+     * @param {FastifyReply} reply
+     */
+    async (request, reply) => {
+      const client = await app.pg.connect()
+      try {
+        await client.query(
+          'UPDATE measurements SET total = total + 1, successful = successful + $1 WHERE subnet = $2',
+          [request.body ? 1 : 0, request.params.subnet]
+        )
+      } finally {
+        client.release()
+      }
+    }
+  )
+
+  app.get(
+    '/:subnet',
```

Comment:
Let's make the path more descriptive please. `GET /walrus` does not communicate what is the expected response.

Looking at https://github.com/CheckerNetwork/spark-stats/blob/main/README.md, I propose the following endpoint name:

```suggestion
    '/:subnet/retrieval-success-rate',
```